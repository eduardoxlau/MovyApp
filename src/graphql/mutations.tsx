import { gql } from '@apollo/client';

export const LOGIN = gql`
  mutation Login($input: LoginInput!) {
    login(input: $input) {
      access_token
      user {
        id
        email
        full_name
        admin
        photo_path
      }
    }
  }
`;
export const CREATE_USER = gql`
  mutation CreateUser($input: CreateUserInput!) {
    createUser(input: $input) {
      id
      email
      full_name
      admin
      photo_path
    }
  }
`;

export const UPDATE_USER = gql`
  mutation UpdateUser($input: UpdateUserInput!) {
    updateUser(input: $input) {
      status
    }
  }
`;

export const REMOVE_USER = gql`
  mutation DeleteUser {
    deleteUser {
      status
    }
  }
`;

export const CREATE_LIST = gql`
  mutation CreateList($input: CreateListInput!) {
    createList(input: $input) {
      id
      name
      description
    }
  }
`;

export const EDIT_LIST = gql`
  mutation UpdateList($input: UpdateListInput!) {
    updateList(input: $input) {
      status
    }
  }
`;

export const REMOVE_LIST = gql`
  mutation DeleteList($id: Float!) {
    deleteList(id: $id) {
      status
    }
  }
`;

export const REMOVE_MOVIE_LIST = gql`
  mutation RemoveMovieToList($input: ListMovieInput!) {
    removeMovieToList(input: $input) {
      status
    }
  }
`;

export const ADD_MOVIE_LIST = gql`
  mutation AddMovieToList($input: ListMovieInput!) {
    addMovieToList(input: $input) {
      status
    }
  }
`;

export const SEEN_MOVIE = gql`
  mutation seenMovie($movieId: Float!) {
    seenMovie(movieId: $movieId) {
      status
    }
  }
`;
