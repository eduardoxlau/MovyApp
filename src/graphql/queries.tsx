import { gql } from '@apollo/client';

export const GET_USER = gql`
  query GetUser {
    getUser {
      id
      email
      full_name
      admin
      lists {
        description
        movies {
          id
          title
        }
      }
    }
  }
`;

export const GET_MOVIES_SEEN = gql`
  query GetMoviesSeen {
    getMoviesSeen {
      id
      title
      overview
      trailer_url
      card_url
      poster_url
      backdrop_url
      stars
      genres {
        name
      }
    }
  }
`;

export const GET_MOVIE = gql`
  query GetMovie($id: Float!) {
    getMovie(id: $id) {
      id
      title
      overview
      trailer_url
      card_url
      poster_url
      backdrop_url
      stars
      genres {
        name
      }
    }
  }
`;

export const GET_MOVIES = gql`
  query GetMovies {
    getMovies(limit: 20) {
      items {
        id
        title
        overview
        trailer_url
        card_url
        poster_url
        backdrop_url
        stars
        genres {
          name
        }
      }
    }
  }
`;

export const GET_MOVIES_PAGINATE = gql`
  query GetMovies($page: Float, $limit: Float) {
    getMovies(page: $page, limit: $limit) {
      meta {
        totalPages
        currentPage
      }
      items {
        id
        title
        overview
        trailer_url
        card_url
        poster_url
        backdrop_url
        stars
        genres {
          name
        }
      }
    }
  }
`;

export const GET_LISTS = gql`
  query GetLists {
    getLists {
      id
      name
      description
      movies {
        id
        title
        trailer_url
        card_url
      }
    }
  }
`;
