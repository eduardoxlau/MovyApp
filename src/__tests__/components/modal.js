/* eslint-disable import/no-unresolved */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import wait from 'waait';
import renderer from 'react-test-renderer';
import { render, fireEvent, act } from '@testing-library/react';

import { MockedProvider } from '@apollo/react-testing';

import { GET_LISTS } from 'graphql/queries';
import { EDIT_LIST } from 'graphql/mutations';
import Modal from 'components/modal';

const list = { id: 1, name: 'test', description: 'test' };
const mocks = [
  {
    request: {
      query: EDIT_LIST,
      variables: { input: list },
    },
    result: { data: { createList: list } },
  },
  {
    request: {
      query: GET_LISTS,
    },
    result: { data: { getLists: [list] } },
  },
];

describe('<Modal />', () => {
  it('should render snapshot Modal', () => {
    const tree = renderer
      .create(
        <MockedProvider mocks={mocks}>
          <Modal />
        </MockedProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should submit form', async () => {
    const onClose = jest.fn();
    const { getByText } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Modal onClose={onClose} edit list={list} />
      </MockedProvider>
    );
    fireEvent.click(getByText(/Save/i));
    await wait(200);
    expect(onClose).toHaveBeenCalled();
  });
});
