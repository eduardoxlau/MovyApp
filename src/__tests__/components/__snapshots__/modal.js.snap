// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render snapshot Modal 1`] = `
<div
  className="w-full h-screen flex items-center fixed"
>
  <div
    className="z-30 bg-black w-full md:w-1/2  min-h-80 m-auto text-white p-10 relative"
  >
    <img
      alt=""
      className="w-8 absolute top-2 right-2 cursor-pointer"
      onClick={[Function]}
      src="close_window.png"
    />
    <form
      className="flex flex-col"
      onSubmit={[Function]}
    >
      <div
        className="text-3xl text-center"
      >
        Create 
         List
      </div>
      <div>
        <div>
          Name:
        </div>
        <input
          className="rounded w-full h-10 my-1.5 text-black indent-1"
          name="name"
          onChange={[Function]}
          placeholder=""
          required={true}
          type="text"
          value=""
        />
      </div>
      <div>
        <div>
          Description:
        </div>
        <input
          className="rounded w-full h-10 my-1.5 text-black indent-1"
          name="description"
          onChange={[Function]}
          placeholder=""
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="flex items-center mt-2"
      >
        <div>
          Is public?
        </div>
        <input
          className="ml-5 w-8 h-8 order-1 md:order-none"
          name="isKid"
          type="checkbox"
        />
      </div>
      <div
        className="mt-5"
      >
        <input
          className="w-full h-11 text-center rounded cursor-pointer bg-button"
          disabled={false}
          type="submit"
          value="Save"
        />
      </div>
    </form>
  </div>
  <div
    className="z-20 absolute opacity-60 bg-black w-full h-full"
  />
</div>
`;
